diff --git a/ash/public/cpp/accelerators.h b/ash/public/cpp/accelerators.h
index 988f7146ec9e3..07fafec4e835f 100644
--- a/ash/public/cpp/accelerators.h
+++ b/ash/public/cpp/accelerators.h
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -244,15 +244,15 @@ ASH_PUBLIC_EXPORT inline constexpr auto kAcceleratorData = std::to_array<
     {true, ui::VKEY_9, ui::EF_ALT_DOWN, AcceleratorAction::kLaunchLastApp},
 
     // Window management shortcuts.
-    {true, ui::VKEY_OEM_4, ui::EF_ALT_DOWN,
+    {true, ui::VKEY_LEFT, ui::VKEY_LWIN,
      AcceleratorAction::kWindowCycleSnapLeft},
-    {true, ui::VKEY_OEM_6, ui::EF_ALT_DOWN,
+    {true, ui::VKEY_RIGHT, ui::VKEY_LWIN,
      AcceleratorAction::kWindowCycleSnapRight},
-    {true, ui::VKEY_OEM_MINUS, ui::EF_ALT_DOWN,
+    {true, ui::VKEY_DOWN, ui::VKEY_LWIN,
      AcceleratorAction::kWindowMinimize},
-    {true, ui::VKEY_F, ui::EF_ALT_DOWN | ui::EF_COMMAND_DOWN,
+    {true, ui::VKEY_OEM_MINUS, ui::EF_ALT_DOWN | ui::EF_COMMAND_DOWN,
      AcceleratorAction::kToggleFloating},
-    {true, ui::VKEY_OEM_PLUS, ui::EF_ALT_DOWN,
+    {true, ui::VKEY_UP, ui::VKEY_LWIN,
      AcceleratorAction::kToggleMaximized},
     {true, ui::VKEY_BROWSER_FORWARD, ui::EF_CONTROL_DOWN,
      AcceleratorAction::kFocusNextPane},
diff --git a/chrome/browser/ui/cocoa/accelerators_cocoa.mm b/chrome/browser/ui/cocoa/accelerators_cocoa.mm
index f7a207457997b..37a33c0e835db 100644
--- a/chrome/browser/ui/cocoa/accelerators_cocoa.mm
+++ b/chrome/browser/ui/cocoa/accelerators_cocoa.mm
@@ -1,4 +1,4 @@
-// Copyright 2011 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -73,6 +73,11 @@ const struct AcceleratorMapping {
     // We used to define IDC_CLOSE_WINDOW here. Instead, see
     // AcceleratorForCloseWindow().
     {IDC_CLOSE_TAB, ui::EF_COMMAND_DOWN, ui::VKEY_W},
+    // Thorium Accelerators
+    {IDC_MOVE_TAB_TO_NEW_WINDOW, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
+     ui::VKEY_P},
+    {IDC_NEW_TAB_TO_RIGHT, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
+     ui::VKEY_K},
 
     {IDC_EMAIL_PAGE_LOCATION, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
      ui::VKEY_I},
@@ -99,7 +104,7 @@ const struct AcceleratorMapping {
     {IDC_HOME, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN, ui::VKEY_H},
     {IDC_BACK, ui::EF_COMMAND_DOWN, ui::VKEY_OEM_4},
     {IDC_FORWARD, ui::EF_COMMAND_DOWN, ui::VKEY_OEM_6},
-    {IDC_BOOKMARK_ALL_TABS, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
+    {IDC_DUPLICATE_TAB, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN, // Changed from IDC_BOOKMARK_ALL_TABS for Thorium
      ui::VKEY_D},
     {IDC_MINIMIZE_WINDOW, ui::EF_COMMAND_DOWN, ui::VKEY_M},
     {IDC_SELECT_NEXT_TAB, ui::EF_CONTROL_DOWN, ui::VKEY_TAB},
diff --git a/chrome/browser/ui/views/accelerator_table.cc b/chrome/browser/ui/views/accelerator_table.cc
index 9a52cf832f28e..67948411837ef 100644
--- a/chrome/browser/ui/views/accelerator_table.cc
+++ b/chrome/browser/ui/views/accelerator_table.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -53,7 +53,10 @@ const AcceleratorMapping kAcceleratorMap[] = {
 #if !BUILDFLAG(IS_MAC)
     {ui::VKEY_D, ui::EF_PLATFORM_ACCELERATOR, IDC_BOOKMARK_THIS_TAB},
     {ui::VKEY_D, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
-     IDC_BOOKMARK_ALL_TABS},
+     IDC_DUPLICATE_TAB},
+    {ui::VKEY_P, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR, IDC_MOVE_TAB_TO_NEW_WINDOW},
+    {ui::VKEY_K, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR, IDC_NEW_TAB_TO_RIGHT},
+    {ui::VKEY_Q, ui::EF_SHIFT_DOWN | ui::EF_CONTROL_DOWN, IDC_EXIT},
     {ui::VKEY_W, ui::EF_PLATFORM_ACCELERATOR, IDC_CLOSE_TAB},
     {ui::VKEY_W, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
      IDC_CLOSE_WINDOW},
@@ -225,7 +228,7 @@ const AcceleratorMapping kAcceleratorMap[] = {
     {ui::VKEY_LEFT, ui::EF_ALT_DOWN, IDC_BACK},
     {ui::VKEY_LEFT, ui::EF_ALTGR_DOWN, IDC_BACK},
 #if BUILDFLAG(ENABLE_PRINTING)
-    {ui::VKEY_P, ui::EF_SHIFT_DOWN | ui::EF_CONTROL_DOWN, IDC_BASIC_PRINT},
+    {ui::VKEY_P, ui::EF_ALT_DOWN | ui::EF_PLATFORM_ACCELERATOR, IDC_BASIC_PRINT},
 #endif  // ENABLE_PRINTING
     {ui::VKEY_B, ui::EF_SHIFT_DOWN | ui::EF_ALT_DOWN, IDC_FOCUS_BOOKMARKS},
     {ui::VKEY_A, ui::EF_SHIFT_DOWN | ui::EF_ALT_DOWN,
@@ -313,11 +316,11 @@ std::vector<AcceleratorMapping> GetAcceleratorList() {
     // Doing this check here and not at the bottom since kUIDebugAcceleratorMap
     // contains Ctrl+Alt keys but we don't enable those for the public.
 #if DCHECK_IS_ON()
-    constexpr int kCtrlAlt = ui::EF_CONTROL_DOWN | ui::EF_ALT_DOWN;
+    /* constexpr int kCtrlAlt = ui::EF_CONTROL_DOWN | ui::EF_ALT_DOWN;
     for (auto& mapping : *accelerators) {
       DCHECK((mapping.modifiers & kCtrlAlt) != kCtrlAlt)
           << "Accelerators with Ctrl+Alt are reserved by Windows.";
-    }
+    } */
 #endif
 
     if (base::FeatureList::IsEnabled(features::kUIDebugTools)) {
